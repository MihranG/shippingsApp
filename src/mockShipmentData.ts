import {IShipment} from './interfaces'

export const data: IShipment[]  =
    [{
        "id": "S1000",
        "name": "TEST1",
        "cargo": [
          {
            "type": "Fabric",
            "description": "1000 Blue T-shirts",
            "volume": "2"
          },
          {
            "type": "Fabric",
            "description": "2000 Green T-shirts",
            "volume": "3"
          }
        ],
        "mode": "sea",
        "type": "FCL",
        "destination": "Saarbrücker Str. 38, 10405 Berlin",
        "origin": "Shanghai Port",
        "services": [
          {
            "type": "customs"
          }
        ],
        "total": "1000",
        "status": "ACTIVE",
        "userId": "U1000"
      },
      {
        "id": "S1001",
        "name": "TEST2",
        "cargo": [
          {
            "type": "Furniture",
            "description": "300 Tables",
            "volume": "20"
          },
          {
            "type": "Furniture",
            "description": "1500 Chairs",
            "volume": "15"
          }
        ],
        "mode": "sea",
        "type": "FCL",
        "destination": "Saarbrücker Str. 38, 10405 Berlin",
        "origin": "Ningbo port",
        "services": [
          {
            "type": "customs"
          },
          {
            "type": "insurance",
            "value": "100"
          }
        ],
        "total": "3000",
        "status": "ACTIVE",
        "userId": "U1002"
      }
  ]

export default data