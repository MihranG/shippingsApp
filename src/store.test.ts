import configureStore from "redux-mock-store";
import {
  setLoading,
  setData,
  store as realStore,
  shipmentsReducer,
  setShipments,
  editShipmentWithId
} from "./store";

import {data as mockShipmentData} from './mockShipmentData';

describe("Redux store", () => {

  describe("should have appropriate", () => {
    const mockStore = configureStore();
    const store = mockStore();
    afterEach(() => {
      store.clearActions();
    });

    test("setLoading action", () => {
      store.dispatch(setLoading(true));
      expect(store.getActions()[0]).toEqual({
        payload: true,
        type: "shipments/setLoading"
      });
    });

    test("setShipments action", () => {
      store.dispatch(setShipments(mockShipmentData));
      expect(store.getActions()[0]).toEqual({
        payload: mockShipmentData,
        type: "shipments/setShipments"
      });
      expect(store.getActions()[0]).toMatchSnapshot();
    });


    test('editShipmentWithId action', ()=>{
        store.dispatch(editShipmentWithId({id: 1, shipmentDetails: {data: 'data'}}))
        expect(store.getActions()[0]).toMatchSnapshot();
        expect(store.getActions()[0]).toEqual({
            payload: {id: 1, shipmentDetails: {data: 'data'}},
            type: "shipments/editShipmentWithId"
        })
    })

  });

  describe("should behave appropriate", () => {
    const initialState = realStore.getState();
    describe("when called timePeriodReducer with", () => {
      const ownInitialState = initialState.shipments;
      test("setLoading action", () => {
        expect(ownInitialState.isLoading).toBe(false);
        const newState = shipmentsReducer(ownInitialState, setLoading(true));
        expect(newState.isLoading).toBe(true);
      });

      describe("setShipments action", () => {
        let newState, newStateFirstIndex, newStateFirstEntity;
        beforeAll(()=>{
            expect(ownInitialState.data).toEqual({});
            newState = shipmentsReducer(
                ownInitialState,
                setShipments(mockShipmentData)
            );
            newStateFirstIndex = Object.keys(newState.data)[0];
            newStateFirstEntity = newState.data[Object.keys(newState.data)[0]]
        })
        
        test('which normalizes given array',()=>{
            
            expect(Array.isArray(newState.data)).toBe(false);
            expect(newStateFirstIndex).toBe(mockShipmentData[0].id)
        })
    
        test('returns the right shipments state', ()=>{
            expect(newStateFirstEntity).toEqual(mockShipmentData[0]);
        })
      });
    });
  });
});