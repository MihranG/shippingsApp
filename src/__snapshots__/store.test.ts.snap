// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux store should have appropriate editShipmentWithId action 1`] = `
Object {
  "payload": Object {
    "id": 1,
    "shipmentDetails": Object {
      "data": "data",
    },
  },
  "type": "shipments/editShipmentWithId",
}
`;

exports[`Redux store should have appropriate setShipments action 1`] = `
Object {
  "payload": Array [
    Object {
      "cargo": Array [
        Object {
          "description": "1000 Blue T-shirts",
          "type": "Fabric",
          "volume": "2",
        },
        Object {
          "description": "2000 Green T-shirts",
          "type": "Fabric",
          "volume": "3",
        },
      ],
      "destination": "Saarbrücker Str. 38, 10405 Berlin",
      "id": "S1000",
      "mode": "sea",
      "name": "TEST1",
      "origin": "Shanghai Port",
      "services": Array [
        Object {
          "type": "customs",
        },
      ],
      "status": "ACTIVE",
      "total": "1000",
      "type": "FCL",
      "userId": "U1000",
    },
    Object {
      "cargo": Array [
        Object {
          "description": "300 Tables",
          "type": "Furniture",
          "volume": "20",
        },
        Object {
          "description": "1500 Chairs",
          "type": "Furniture",
          "volume": "15",
        },
      ],
      "destination": "Saarbrücker Str. 38, 10405 Berlin",
      "id": "S1001",
      "mode": "sea",
      "name": "TEST2",
      "origin": "Ningbo port",
      "services": Array [
        Object {
          "type": "customs",
        },
        Object {
          "type": "insurance",
          "value": "100",
        },
      ],
      "status": "ACTIVE",
      "total": "3000",
      "type": "FCL",
      "userId": "U1002",
    },
  ],
  "type": "shipments/setShipments",
}
`;
